nav2_mppi_controller/include/nav2_mppi_controller/tools/utils.hpp

```diff
@@ -510,8 +510,71 @@ inline void savitskyGolayFilter(
     control_sequence.wz(offset)};
 }
 
+/**
+ * @brief Find the first pose at which there is an in-place rotation in the path
+ * @param path Path to check
+ * @param translation_threshold Maximum translation per pose transition
+ * @param rotation_threshold Minimum total rotation to consider significant
+ * @return Index after the rotation sequence if found, path size if no rotation detected
+ */
+inline unsigned int findFirstPathRotation(
```

Comment:
For consideration: Do we really need this to be another function? We could just have this be a possible inversion check on `findFirstPathInversion` with a bool for `check_rotations` (probably a better argument name, but you get the idea). If so, we check if there's an inplace rotation to return.